lib/utils/notification_service.dart

```diff
@@ -130,43 +138,204 @@ class NotificationService {
+        notificationId,
+        'notificationTitle'.tr,
+        'notificationBody'.tr,
+        tz.TZDateTime.parse(tz.local, dateTime.toString()),
+        _platformNotificationDetails,
+        uiLocalNotificationDateInterpretation:
+            UILocalNotificationDateInterpretation.absoluteTime,
+        androidScheduleMode: AndroidScheduleMode.exactAllowWhileIdle,
+      );
+      notificationDates.add(OSDDateTime(
+        year: dateTime.year,
+        month: dateTime.month,
+        day: dateTime.day,
+        hour: dateTime.hour,
+        minute: dateTime.minute,
+        notificationId: notificationId,
+      ));
+
+      // increase day
+      dateTime = dateTime.add(const Duration(days: 1));
+      notificationId = getNotificationId();
+    }
+
+    // saving notification dates in db
+    Utils.saveDateTimes(notificationDates);
+    Utils.logInfo(
+      '[NOTIFICATIONS] - Notifications were scheduled',
+    );
+  }
+
+  void rescheduleNotifications(int hour, int minute) async {
+    if (!isNotificationActivated()) return;
+
+    int notificationId = getNotificationId();
+
     _flutterLocalNotificationsPlugin.cancelAll();
 
-    // sets the scheduled time in DateTime format
-    final String setTime = DateTime(
-      day.year,
-      day.month,
-      day.day,
+    // set scheduled time in DateTime format
+    final DateTime today = DateTime.now();
+    DateTime dateTime = DateTime(
+      today.year,
+      today.month,
+      today.day,
       hour,
       minute,
-    ).toString();
+    );
+
+    final List<OSDDateTime> notificationDates = [];
 
-    Utils.logInfo('[NOTIFICATIONS] - Scheduled with setTime=$setTime');
+    // schedule notifications
+    while (notificationDates.length < Constants.scheduleNotificationForDays) {
+      if (dateTime.isAfter(DateTime.now())) {
+        await _flutterLocalNotificationsPlugin.zonedSchedule(
+          notificationId,
+          'notificationTitle'.tr,
+          'notificationBody'.tr,
+          tz.TZDateTime.parse(tz.local, dateTime.toString()),
+          _platformNotificationDetails,
+          uiLocalNotificationDateInterpretation:
+              UILocalNotificationDateInterpretation.absoluteTime,
+          androidScheduleMode: AndroidScheduleMode.exactAllowWhileIdle,
+        );
+        notificationDates.add(OSDDateTime(
+          year: dateTime.year,
+          month: dateTime.month,
+          day: dateTime.day,
+          hour: dateTime.hour,
+          minute: dateTime.minute,
+          notificationId: notificationId,
+        ));
+        notificationId = getNotificationId();
+      }
 
-    /// Schedule notification
+      // increase day
+      dateTime = dateTime.add(const Duration(days: 1));
+    }
+
+    // saving notification dates in db
+    Utils.saveDateTimes(notificationDates);
+    Utils.logInfo(
+      '[NOTIFICATIONS] - Notifications were rescheduled',
+    );
+  }
+
+  void cancelTodayNotification() async {
```

Comment:
@ishanvaghani This is not working properly, this method is executed when adding a video for today, but the notification shows up (when it should be cancelled).